src/QsCompiler/Tests.Compiler/LambdaLiftingTests.fs

```diff
@@ -0,0 +1,573 @@
+
+    [<Fact>]
+    [<Trait("Category", "Return Values")>]
+    member this.``With Type Parameters``() =
+        let testNumber = 11
+        let srcChunks = ReadAndChunkSourceFile "LambdaLifting.qs"
+        srcChunks.Length >= testNumber + 1 |> Assert.True
+        let shared = srcChunks.[0]
+        let compilationDataStructures = BuildContent <| shared + srcChunks.[testNumber]
+        let result = LiftLambdaExpressions.Apply compilationDataStructures.BuiltCompilation
+        Assert.NotNull result
+
+        let generated =
+            GetCallablesWithSuffix result Signatures.LambdaLiftingNS "_Foo"
+            |> (fun x ->
+                Assert.True(1 = Seq.length x)
+                Seq.item 0 x)
+
+        let originalExpectedName = { Namespace = Signatures.LambdaLiftingNS; Name = "Foo" }
+        let ``Foo.A`` = QsTypeParameter.New(originalExpectedName, "A") |> TypeParameter |> ResolvedType.New
+        let ``Foo.B`` = QsTypeParameter.New(originalExpectedName, "B") |> TypeParameter |> ResolvedType.New
+        let ``Foo.C`` = QsTypeParameter.New(originalExpectedName, "C") |> TypeParameter |> ResolvedType.New
+
+        let originalExpectedArgType =
+            [| ``Foo.A``; ``Foo.B``; ``Foo.C`` |]
+            |> ImmutableArray.ToImmutableArray
+            |> QsTypeKind.TupleType
+            |> ResolvedType.New
+
+        let originalExpectedReturnType = ResolvedType.New(ResolvedTypeKind.UnitType)
+
+        let originalSigExpected = originalExpectedName, originalExpectedArgType, originalExpectedReturnType
+
+        let generatedExpectedName = { Namespace = Signatures.LambdaLiftingNS; Name = generated.FullName.Name }
+        let ``_Foo.A`` = QsTypeParameter.New(generatedExpectedName, "A") |> TypeParameter |> ResolvedType.New
+        let ``_Foo.C`` = QsTypeParameter.New(generatedExpectedName, "C") |> TypeParameter |> ResolvedType.New
+
+        let generatedExpectedArgType =
+            [| ``_Foo.A``; ``_Foo.C``; ResolvedType.New(ResolvedTypeKind.UnitType) |]
+            |> ImmutableArray.ToImmutableArray
+            |> QsTypeKind.TupleType
+            |> ResolvedType.New
+
+        let generatedExpectedReturnType =
+            [| ``_Foo.C``; ``_Foo.A`` |]
+            |> ImmutableArray.ToImmutableArray
+            |> QsTypeKind.TupleType
+            |> ResolvedType.New
+
+        let generatedSigExpected = generatedExpectedName, generatedExpectedArgType, generatedExpectedReturnType
+
+        Signatures.SignatureCheck
+            [ Signatures.LambdaLiftingNS ]
+            (seq {
+                originalSigExpected
+                generatedSigExpected
+             })
+            result
+
+    [<Fact>]
+    [<Trait("Category", "Return Values")>]
+    member this.``With Nested Lambda Call``() = CompileLambdaLiftingTest 12
+
+    [<Fact>]
+    [<Trait("Category", "Return Values")>]
+    member this.``With Nested Lambda``() = CompileLambdaLiftingTest 13
```

Comment:
Where are the assertions for these cases?

Reply:
That the result from compilation has the appropriate signatures is enough for this test case. That is handled in the signature check.

Reply:
resolved, check [this comment](https://github.com/microsoft/qsharp-compiler/pull/1302#discussion_r792346904)